File: src/com/lysdev/transperthcached/timetable/VisitComparator.java
package com.lysdev.transperthcached.timetable;

import java.util.Comparator;

public class VisitComparator implements Comparator<Visit> {
    public int compare(Visit v1, Visit v2) {
        if      (v1.time.isBefore(v2.time)) return  1;
        else if (v1.time.isAfter(v2.time))  return -1;
        else                           return  0;
    }
}

File: src/com/lysdev/transperthcached/timetable/VisitComparator.scala
package com.lysdev.transperthcached.timetable

import java.util.Comparator

class VisitComparator extends Comparator[Visit] {
    def compare(v1: Visit, v2: Visit) : Int = {
        if      (v1.getTime.isBefore(v2.getTime)) -1
        else if (v1.getTime.isAfter(v2.getTime))   1
        else                                       0
    }
}
